src/test/java/com/weather/discordweather/WeatherControllerTest.java

```diff
@@ -136,9 +136,61 @@ public void getWeatherValid() {
           geocodeResponse
       );
       assertThat(forecast).isNotEmpty();
-      Mockito.when(gateway.getWeatherForecast(33.0, 80.0)).thenReturn(forecast);
+      Mockito.when(gateway.getWeatherForecast(33.0, 80.0, "")).thenReturn(forecast);
       var controller = new WeatherController(gateway);
-      var response = controller.weather(33.0, 80.0);
+      var response = controller.weather(33.0, 80.0, "");
+      assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+      assertThat(response.getBody()).isEqualTo(forecast.get());
+    }
+
+    @Test
+    @DisplayName("returns a 400 Bad Request if the coordinates and location are missing")
+    public void getWeatherMissingCoordinatesAndLocation() {
+      var response = new WeatherController(gateway).weather(null, null, "");
+      Mockito.verifyNoInteractions(gateway);
+      assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @DisplayName("returns a 200 OK if location is valid")
```

Comment:
Is there a test case for if all three of `lat`, `lon`, and `location` are provided?

Reply:
There is not. The location is checked first though. Should we prioritize lat/lon or location?

Reply:
Up to you on this one. I can hear arguments for both sides. (...Or alternately, ban having both because it's ambiguous.)

Whichever you pick though, I'd document that with a test showing it.